fili-core/src/main/java/com/yahoo/bard/webservice/table/ConcretePhysicalTable.java

```diff
@@ -20,25 +22,51 @@
 
     /**
      * Create a concrete physical table.
-     * The availability on this table is initialized to empty intervals.
      *
-     * @param name  Name of the physical table as TableName
+     * @param name  Name of the physical table as TableName, also used as fact table name
      * @param timeGrain  time grain of the table
      * @param columns  The columns for this table
      * @param logicalToPhysicalColumnNames  Mappings from logical to physical names
+     * @param metadataService  Datasource metadata service containing availability data for the table
      */
     public ConcretePhysicalTable(
             @NotNull TableName name,
             @NotNull ZonedTimeGrain timeGrain,
-            @NotNull Iterable<Column> columns,
-            @NotNull Map<String, String> logicalToPhysicalColumnNames
+            @NotNull Set<Column> columns,
+            @NotNull Map<String, String> logicalToPhysicalColumnNames,
+            @NotNull DataSourceMetadataService metadataService
+    ) {
+        this(
+                name,
+                timeGrain,
+                columns,
+                logicalToPhysicalColumnNames,
+                new ConcreteAvailability(name, columns, metadataService)
+        );
+    }
+
+    /**
+     * Create a concrete physical table, the availability on this table is built externally.
+     *
+     * @param name  Name of the physical table as TableName, also used as fact table name
+     * @param timeGrain  time grain of the table
+     * @param columns  The columns for this table
+     * @param logicalToPhysicalColumnNames  Mappings from logical to physical names
+     * @param availability  Availability that serves interval availability for columns
+     */
+    public ConcretePhysicalTable(
+            @NotNull TableName name,
+            @NotNull ZonedTimeGrain timeGrain,
+            @NotNull Set<Column> columns,
+            @NotNull Map<String, String> logicalToPhysicalColumnNames,
+            @NotNull Availability availability
```

Comment:
Yes, I think so. I can't think of a reason for someone to have a ConcretePhysicalTable that does not also have a ConcreteAvailability, since the _primary_ contract of a ConcreteAvailability is that it is providing availability info for a _single_ backing data source.